packages/grid/x-grid/src/tests/filtering.XGrid.test.tsx

```diff
@@ -224,4 +226,85 @@ describe('<XGrid /> - Filter', () => {
     setProps({ filterModel: { items: [] } });
     expect(getColumnValues()).to.deep.equal(['Nike', 'Adidas', 'Puma']);
   });
+
+  describe('Server', () => {
+    it('should refresh the filter panel when adding filters', () => {
+      function loadServerRows(commodityFilterValue) {
+        const serverRows = [
+          { id: '1', commodity: 'rice' },
+          { id: '2', commodity: 'soybeans' },
+          { id: '3', commodity: 'milk' },
+          { id: '4', commodity: 'wheat' },
+          { id: '5', commodity: 'oats' },
+        ];
+
+        return new Promise<RowModel[]>((resolve) => {
+          setTimeout(() => {
+            if (!commodityFilterValue) {
+              resolve(serverRows);
+              return;
+            }
+            resolve(
+              serverRows.filter(
+                (row) => row.commodity.toLowerCase().indexOf(commodityFilterValue) > -1,
+              ),
+            );
+          }, Math.random() * 500 + 100); // simulate network latency
```

Comment:
This Math.random() and setTimeout are red flags. Please simplify the test case as much as possible to nail the root cause (failing on master). We don't need the ceremony.